packages/p/pahomqttc/xmake.lua

```diff
@@ -0,0 +1,25 @@
+package("pahomqttc")
+    set_homepage("https://github.com/eclipse/paho.mqtt.c")
+    set_description("Eclipse Paho MQTT C Client Library")
+    set_license("MIT")
+
+    add_urls("https://github.com/eclipse/paho.mqtt.c/archive/refs/tags/$(version).zip",
+             "https://github.com/eclipse/paho.mqtt.c.git")
+
+    add_versions("v1.3.13", "5ba7c7ab7ebb1499938fa2e358e6c1f9a926b270f2bf082acf89d59b4771a132")
+
+    add_deps("cmake")
+
+    on_install(function (package)
+        local configs = {"-DPAHO_BUILD_SAMPLES=FALSE",
+                         "-DPAHO_WITH_SSL=FALSE",
+                         "-DPAHO_BUILD_DOCUMENTATION=FALSE"}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        assert(package:has_cxxtypes("MQTTClient_message", {includes = "MQTTClient.h"}))
```

Comment:
is this c library? you can use has_cfuncs to check symbol links

Reply:
Yes it is a C lib. 

see #[682661b](https://github.com/xmake-io/xmake-repo/pull/2757/commits/682661b996cff7df110718494fa34bd1fc8288e1)

Thank you for the reviews, it's all new to me. I appreciate it.